train_gpt2.py

```diff
@@ -505,7 +505,8 @@ def get_lr(it):
     for param_group in optimizer.param_groups:
         param_group['lr'] = lr
     optimizer.step()
-    torch.cuda.synchronize() # wait for the GPU to finish work
+    if(device.startswith("cuda")):
```

Comment:
not proper python, do not use brackets.
also, maybe use device_type == 'cuda' instead?

Reply:
keeping consistent with:

https://github.com/karpathy/build-nanogpt/blob/6104ab1b53920f6e2159749676073ff7d815c1fa/train_gpt2.py#L316

Reply:
on bracket's I'm not really python dev to argue, but this code works instead of blowing up on cpu - profit ;)

Reply:
Ok I see you fixed it yourself, so thank